xapian-core/backends/dbfactory.cc

```diff
@@ -268,7 +272,7 @@ Database::Database(const string& path, int flags)
 	throw FeatureUnavailableError("Flint backend no longer supported");
     }
 
-    throw DatabaseOpeningError("Couldn't detect type of database");
+    throw DatabaseNotFoundError("Couldn't detect type of database");
```

Comment:
If we get to here `path` exists and is a directory.

What's you reasoning for this case being "not found"?

And if this is "not found", why not the case around line 229/233 above where `path` exists but isn't a regular file or directory?

Reply:
The reasoning of when it should return `DatabaseNotFoundError` instead of `DatabaseOpeningError` is, I believe, simple. If you try `DB_OPEN` and it fails, the failure should be:

1. `DatabaseNotFoundError` if passing `DB_CREATE` or `DB_OPEN_OR_CREATE` or `DB_CREATE_OR_OVERWRITE` on the same `path` would succeed.

2. `DatabaseOpeningError` if passing `DB_CREATE` or `DB_OPEN_OR_CREATE` or `DB_CREATE_OR_OVERWRITE` on the same `path` would still fail.

Reply:
That reminds me that we discussed this before, and looking back in addition to the rule above we both thought you should also get `DatabaseNotFoundError` if the parent directories don't all exist: https://github.com/xapian/xapian/pull/213#issuecomment-450450384

So I guess the rule is if `mkdir -p path` followed by opening a `WritableDatabase` on `path` with `DB_CREATE` etc would succeed (or be expected to - I don't think we want to get into checking permissions, etc).

Anyway, this still seems logical (if we make the obvious allowances for read-only backends like current honey).